imaginairy/schema.py

```diff
@@ -172,385 +133,543 @@ def handle_b64(value: Any) -> "LazyLoadingImage":
+    #     from imaginairy.samplers import SAMPLER_LOOKUP
+
+    #     if v is None:
+    #         SamplerCls = SAMPLER_LOOKUP[info.data["sampler_type"]]
+    #         v = SamplerCls.default_steps
+
+    #     return int(v)
+
+    # @model_validator(mode="after")
+    # def validate_init_image_strength(self):
+    #     if self.init_image_strength is None:
+    #         if self.control_inputs:
+    #             self.init_image_strength = 0.0
+    #         elif self.outpaint or self.mask_image or self.mask_prompt:
+    #             self.init_image_strength = 0.0
+    #         else:
+    #             self.init_image_strength = 0.2
+
+    #     return self
+
+
+@dataclass
+class ImaginePrompt:
+    prompt: list[WeightedPrompt]
+    negative_prompt: list[WeightedPrompt]
+    prompt_strength: float
+    init_image: InitImageConfig | None
+    width: int
+    height: int
+    seed: int
+    steps: int
+    model: str
+    model_config: Path  # ?: what is this?
+    sampler_type: SamplerType
+    control_inputs: list[ControlNetInput]
+    outpaint: str  # TODO: implement OutpaintConfig class
+    face_fixer: FaceFixerConfig
+    tiling: TileMode
+    caption_text: str | None
+
+    def __init__(
+        self,
+        prompt: PromptInput | None = None,
+        negative_prompt: PromptInput | None = None,
+        prompt_strength: float = 7.5,
+        init_image: InitImageInput | None = None,
+        width: int | None = None,
+        height: int | None = None,
+        seed: int | None = None,
+        steps: int = 30,
+        model: str = config.DEFAULT_MODEL,
+        sampler_type: str = config.DEFAULT_SAMPLER,
+        control_inputs: list[ControlNetInput] | None = None,
+        outpaint: str = "",
+        face_fixer: FaceFixerConfig | float | bool | None = None,
+        tiling: TileMode | str | bool | None = None,
+        caption_text: str | None = None,
+    ) -> None:
+        """
+        Args:
+            prompt:
+                Text prompt(s) to condition the model on. Can be a single string
+                or a list of strings.
+            negative_prompt:
+                Text prompt(s) to condition the model on. Can be a single string
+                or a list of strings.
+            prompt_strength:
+                How much to weigh the prompt when sampling. Higher values mean
+                more influence, but can also lead to more artifacts.
+            init_image:
+                Image to initialize the model with. Can be a filepath, url,
+                base64 string, PIL Image, or LazyLoadingImage.
+            width:
+                Width of the generated image in pixels.
+            height:
+                Height of the generated image in pixels.
+            seed:
+                Random seed to use for sampling.
+            steps:
+                Number of sampling steps to perform.
+            model:
+                Name of the model to use. See the list of available models in
```

Comment:
It also accepts a path or a url to weights, which is pretty convenient but a confusing api.  If it's a url it handles the caching.